src/component/DefaultLegendContent.tsx

```diff
@@ -169,6 +170,7 @@ export class DefaultLegendContent extends PureComponent<Props> {
         return null;
       }
 
+      const entryValue = _.isFunction(entry.value) ? entry.value() : entry.value;
```

Comment:
hi, I'm not sure if I understood correctly, but I don't think this is correct fix. the fn that is provided expects an argument which is not provided here - yeah, it would fix the warning but I believe it won't work correct @ckifer don't you agree?
<img width="293" alt="image" src="https://github.com/recharts/recharts/assets/4653046/e6ceda5f-a011-4bf2-a588-bcf1d9d93ade">

Reply:
I'm confused. This should work fine - I'll pull it down later and check

Reply:
shouldn't it be something like
`const entryValue = _.isFunction(entry.value) ? entry.value(entry) : entry.value;`
or
`const entryValue = _.isFunction(entry.value) ? entry.value(entry.payload) : entry.value;` 
?

in given example - in the issue, the reporter set the handler as below:
`row => row.pv`

in this example and current code - what is row? wouldn't it throw 'row is undefined' ?

I might be wrong though.

Reply:
yeah you're right, thanks for taking the extra care to check. Just need to fix this before the next release and should be okay

@chris-mcdonald-dev fyi

Reply:
@MateuszTrN Yeah, thanks for catching that. Looking into a solution now. We won't be able to get it from the entry or payload for the Legend component, since that doesn't seem to contain the chart data and just the payload for the legend elements.

Reply:
we're really just trying to get the actual dataKey name here, not any actual values. When you have `data => data.pv` you don't really have great reference to the text `pv`

Reply:
Good point. I can't see the point of this being a function directly passed to the `Legend` component either. 

I'll spend some time thinking of a solution. The only initial idea I have is rendering an empty string when a function is derived from a chart component's `dataKey` and printing a warning message to the console saying to pass a payload value as a string to `<Legend/>` whenever that's the case.

Otherwise, I'm not sure how we'd have any way of knowing what that legend is supposed to be labeled since functions won't necessarily have to follow the format of `() => someObject.theValue`

Reply:
It does also use the "name" property before the dataKey property in the Legend. If the dataKey is a function we could print a warning that name is required for a label to appear in the legend

Reply:
Ah true! I'll create an issue and open a matching PR with tests in a bit.